### ISSUE TITLE
Error("invalid byte sequence in UTF-8") occurs when commiting message includes multibyte string in Mac OSX.

### ISSUE THREAD

**Issue by keitaoouchi** [brigade/overcommit#328]
When I commit with Japanese characters, following error occured. I have no problem when I commit with ascii characters.

OS: Mac OSX El Capitan (10.11.3)

```
$ git commit -m "あ"
invalid byte sequence in UTF-8
/Users/200219/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/overcommit-0.31.0/lib/overcommit/hook_context/pre_commit.rb:20:in `=~'
/Users/200219/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/overcommit-0.31.0/lib/overcommit/hook_context/pre_commit.rb:20:in `amendment?'
/Users/200219/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/overcommit-0.31.0/lib/overcommit/hook_context/pre_commit.rb:95:in `modified_files'
/Users/200219/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/overcommit-0.31.0/lib/overcommit/hook_context/pre_commit.rb:180:in `store_modified_times'
/Users/200219/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/overcommit-0.31.0/lib/overcommit/hook_context/pre_commit.rb:37:in `setup_environment'
/Users/200219/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/overcommit-0.31.0/lib/overcommit/hook_runner.rb:36:in `block in run'
/Users/200219/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/overcommit-0.31.0/lib/overcommit/interrupt_handler.rb:84:in `isolate_from_interrupts'
/Users/200219/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/overcommit-0.31.0/lib/overcommit/hook_runner.rb:24:in `run'
.git/hooks/pre-commit:79:in `<main>'
Report this bug at https://github.com/brigade/overcommit/issues
```

**Comment by sds** [comment]
This is an issue with how the `ps` command reports the original command line of a process when it contains unicode characters.

You can reproduce it with the following:

```ruby
# test.rb
command = `ps -ocommand= -p #{Process.pid}`
/some-regex/ =~ command
```

```
$ ruby test.rb "あ"
test.rb:2:in `<main>': invalid byte sequence in UTF-8 (ArgumentError)
```

If you look at the output of `ps -ocommand= ...`, you'll see that it escapes the code point with some odd sequence:

```
ruby test.rb ^AM^B
```

Quick digging hasn't turned up anything useful with what these escape sequences are. I'm tempted to work around the problem by encoding the string in a different encoding, replacing invalid characters, and then encoding back to UTF8, allowing us to avoid the issue since the strings we care about in this particular case are going to be regular ASCII characters, so they won't get corrupted.

```ruby
# test_fix.rb
command = `ps -ocommand= -p #{Process.pid}`.chomp

puts "Before: #{command}"
unless command.valid_encoding?
  command = command.encode('UTF-16be', invalid: :replace, replace: '?').encode('UTF-8')
end
puts "After: #{command}"

/some-regex/ =~ command
```

```
$ ruby test_fix.rb "あ"
Before: ruby test.rb ^AM^B
After: ruby test.rb ?M^AM^B
```

@jawshooah Before I go ahead and implement the above hacky workaround, do you have any thoughts since this was added by you in 16509a8f227339be81023365c14260?

**Comment by jawshooah** [comment]
Not sure what else can be done, so I'm fine with it. Was going to suggest `String#scrub`, but I forgot we have to support Rubies older than 2.1.

**Comment by sds** [comment]
Fixed in [027e88b](https://github.com/brigade/overcommit/compare/7624e70...83147b8).
